src/Stratis.Bitcoin.Features.Api/CertificateStore.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.IO;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    public class CertificateStore : ICertificateStore
+    {
+        public bool TryGet(string filePath, out X509Certificate2 certificate)
+        {
+            certificate = null;
+
+            if (!File.Exists(filePath))
+                return false;
+
+            var fileInBytes = File.ReadAllBytes(filePath);
+            certificate = new X509Certificate2(fileInBytes);
```

Comment:
Both of these lines can throw in which case I would expect this method to return false. But you don't have any comments anywhere, so I have no idea what your intention was

Reply:
Intention is to throw, I initially had a try catch rethrow, but then thought it was useless

Reply:
But you throw here https://github.com/stratisproject/StratisBitcoinFullNode/pull/1897/files/74a135a28a70b125505334abfbd8dd0811ba3f45#diff-5257e1acdaea875a7a2d2c6169336535R77

Basically it is quite hard to use this method GetHttpsCertificate bcause you don't declare which exception can be thrown and it looks like you ony throw FileLoadException, but you actually can throw many other exceptions. By using try prefix it looks like it should not throw and it does.